src/test/scala/com/microsoft/hyperspace/index/E2EHyperspaceRulesTests.scala

```diff
@@ -450,7 +450,64 @@ class E2EHyperspaceRulesTests extends HyperspaceSuite with SQLHelper {
 
       // Verify index usage on latest version of index (v=1) after refresh.
       verifyIndexUsage(query2, getIndexFilesPath(indexConfig.indexName, 1))
-      FileUtils.delete(dataPath)
+      FileUtils.delete(new Path(location))
+    }
+  }
+
+  test(
+    "Validate index usage for enforce delete on read " +
+      "after some source data file is deleted.") {
+    withSQLConf(
+      IndexConstants.INDEX_LINEAGE_ENABLED -> "true",
+      IndexConstants.ENFORCE_DELETE_ON_READ_ENABLED -> "true") {
+
+      // Save a copy of source data files.
+      val location = testDir + "deleteOnReadTest"
+      val dataPath = new Path(location, "*parquet")
+      val dataColumns = Seq("c1", "c2", "c3", "c4", "c5")
+      SampleData.save(spark, location, dataColumns)
+
+      // Create index on original source data files.
+      val df = spark.read.parquet(location)
+      val indexConfig = IndexConfig("filterIndex", Seq("c3"), Seq("c1"))
+      hyperspace.createIndex(df, indexConfig)
+
+      // Verify index usage.
+      def query1(): DataFrame =
+        spark.read.parquet(location).filter("c3 == 'facebook'").select("c3", "c1")
+
+      verifyIndexUsage(query1, getIndexFilesPath(indexConfig.indexName))
+
+      // Delete one source data file.
+      val dataFileNames = dataPath
+        .getFileSystem(new Configuration)
+        .globStatus(dataPath)
+        .map(_.getPath)
+
+      assert(dataFileNames.nonEmpty)
+      val fileToDelete = dataFileNames.head
+      FileUtils.delete(fileToDelete)
+
+      def query2(): DataFrame =
+        spark.read.parquet(location).filter("c3 == 'facebook'").select("c3", "c1")
+
+      // Verify index is not used after source data file is deleted.
+      spark.enableHyperspace()
+      val planRootPaths = getAllRootPaths(query2().queryExecution.optimizedPlan)
+      spark.disableHyperspace()
+      assert(planRootPaths.equals(Seq(PathUtils.makeAbsolute(location))))
+
+      // Refresh index to update its metadata.
+      hyperspace.refreshIndex(indexConfig.indexName)
+
+      // Verify index is used after refresh fixes fingerprint in index metadata.
+      // TODO Replace check with verifyIndexUsage() once EnforceDeleteOnRead covers query path.
+      spark.enableHyperspace()
+      assert(
+        queryPlanHasExpectedRootPaths(
+          query2().queryExecution.optimizedPlan,
+          getIndexFilesPath(indexConfig.indexName)))
+      FileUtils.delete(new Path(location))
```

Comment:
Can you utilize `withTempDir`?

Reply:
Actually, we already had `withTempDir` under `HyperspaceSuite` and it served the purpose; I just switched the code to use that.